pkg/server/server.go

```diff
@@ -289,10 +289,14 @@ func (s *Server) HTTPHandler() (http.Handler, error) {
 	authHandlerWithUser := func(h HandlerWithUser) http.HandlerFunc {
 		return authMiddlewareWithUser(authenticator, s.CSRFVerifier, h)
 	}
+
+	tokenReviewHandler := func(h http.HandlerFunc) http.HandlerFunc {
+		return withTokenReview(authenticator, h)
+	}
 	handleFunc(authLoginEndpoint, s.Authenticator.LoginFunc)
 	handleFunc(authLogoutEndpoint, allowMethod(http.MethodPost, s.handleLogout))
 	handleFunc(AuthLoginCallbackEndpoint, s.Authenticator.CallbackFunc(fn))
-	handle(copyLoginEndpoint, authHandler(s.handleCopyLogin))
+	handle(copyLoginEndpoint, authHandler(tokenReviewHandler(s.handleCopyLogin)))
```

Comment:
Lets just use the wrapper in the `authHandle` instead of repetitive use

```go
	authHandler := func(h http.HandlerFunc) http.HandlerFunc {
		return tokenReviewHandler(authMiddleware(authenticator, s.CSRFVerifier, h))
	}
```

Reply:
I nested them in this order bc if we execute token handler middleware before auth middleware, then we fire the token review request before the CSRF check, which is not desirable.

Reply:
I see, could we then pass the `tokenReviewHandler` to the `authMiddleware`, to be executed at the middleware level ?

Reply:
Putting it in authHandler would cause token reviews to be run on endpoints where we don't need it.